src/coreclr/src/vm/methodtable.cpp

```diff
@@ -9890,6 +9870,54 @@ NOINLINE BYTE *MethodTable::GetLoaderAllocatorObjectForGC()
     return retVal;
 }
 
+UINT32 MethodTable::GetNativeSize()
+{
+    CONTRACTL
+    {
+        THROWS;
+        MODE_ANY;
+        PRECONDITION(CheckPointer(GetClass()));
+    }
+    CONTRACTL_END;
+    if (IsBlittable())
+    {
+        return GetClass()->GetLayoutInfo()->GetManagedSize();
+    }
+    return GetNativeLayoutInfo()->GetSize();
+}
+
+EEClassNativeLayoutInfo const* MethodTable::GetNativeLayoutInfo()
+{
+    CONTRACTL
+    {
+        THROWS;
+        MODE_ANY;
+        PRECONDITION(HasLayout());
+    }
+    CONTRACTL_END;
+    EEClassNativeLayoutInfo* pNativeLayoutInfo = GetClass()->GetNativeLayoutInfo();
+    if (pNativeLayoutInfo != nullptr)
+    {
+        return pNativeLayoutInfo;
+    }
+    EnsureNativeLayoutInfoInitialized();
```

Comment:
```suggestion
    return EEClassNativeLayoutInfo::InitializeNativeLayoutFieldMetadataThrowing(this);
```

Reply:
InitializeNativeLayoutFieldMetadataThrowing doesn't return the pointer to the EENativeLayoutInfo . It also doesn't compile for the DAC (which is why the call to it is ifdef'd out when compiling for the dac below).

Reply:
Changing InitializeNativeLayoutFieldMetadataThrowing to return the value would be part of this suggestion.

>  It also doesn't compile for the DAC

It means that this will return NULL under DAC that will be likely followed by null reference exception shortly after. It is better to put `DacNotImpl` into codepaths not implemented for DAC. It gives more predicable results.

Reply:
I can do that! I’ll make sure to put DacNotImpl in there.